python/amici/sbml_import.py

```diff
@@ -1486,28 +1486,45 @@ class TemplateAmici(Template):
     delimiter = 'TPL_'
 
 
-def assignmentRules2observables(sbml_model, filter_function=lambda *_: True):
+def assignmentRules2observables(sbml_model,
+                                filter_function=lambda *_: True,
+                                filter_by_name=False):
     """Turn assignment rules into observables.
 
     Arguments:
         sbml_model: an sbml Model instance
-        filter_function: callback function taking assignment variable as input and returning True/False to indicate if the respective rule should be turned into an observable
+        filter_function: callback function taking assignment variable as input
+        and returning True/False to indicate if the respective rule should be
+        turned into an observable
+        filter_by_name: boolean indicating whether assignment variables are
```

Comment:
We can also just pass the respective sbml parameter object and let the user call getId() or getName() or anything else directly... Breaking change, but barely used yet anyways.